### PR TITLE
Feature/evaluate models

### PR DESCRIPTION
**Pull request by rahelbeloch**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -24,16 +24,16 @@
...
     os.makedirs(LOG_PATH, exist_ok=True)
 
     if model_name not in SUPPORTED_MODELS:
         raise ValueError("Model type '%s' is not supported." % model_name)
 
-    print(f'Configuration:\n model_name: {model_name}\n data_name: {data_name}\n max epochs: {epochs}\n  patience:'
-          f' {patience}\n seed: {seed}\n batch_size: {b_size}\n l_rate_enc: {l_rate_enc}\n l_rate_cl: {l_rate_cl}\n'
-          f' warmup: {warmup}\n weight_decay_enc: {w_decay_enc}\n weight_decay_cl: {w_decay_cl}\n  cf_hidden_dim: '
-          f'{cf_hidden_dim}\n checkpoint: {checkpoint}\n h_search: {h_search}\n GNN layer: {gnn_layer_name}\n')
+    print(f'\nConfiguration:\n mode: {"TEST" if eval else "TRAIN"}\n model_name: {model_name}\n data_name: {data_name}'
```
Comment (rahelbeloch) [inline train.py:33]: I was wanting to put the parameters we have in train and eval mode on the top.

**Review by rahelbeloch — COMMENTED** [review]
I had this code locally, because for some models the evaluation did not go through, so thought this could be useful for everybody.

**Review by matprst — APPROVED** [review]
Looks good

**Review by Matyasch — APPROVED** [review]
Is it so that we can trust the output by putting it into its own script?

**Comment by rahelbeloch** [comment]
> Is it so that we can trust the output by putting it into its own script?

@Matyasch Hmm no, not necessarily. More to be able to only evaluate specific models. When I wasn training I had quite some models which had memory issues right in the end when testing (but after training was already done).

**Pull request merged**